server/channels/app/plugin_requests.go

```diff
@@ -158,96 +188,73 @@ func (ch *Channels) servePluginRequest(w http.ResponseWriter, r *http.Request, h
 	}
 
 	if mfaAppErr := app.MFARequired(rctx); mfaAppErr != nil {
-		pluginID := mux.Vars(r)["plugin_id"]
+		pluginID := params["plugin_id"]
 		ch.srv.Log().Warn("Treating session as unauthenticated since MFA required",
 			mlog.String("plugin_id", pluginID),
 			mlog.String("url", r.URL.Path),
 			mlog.Err(mfaAppErr),
 		)
-		token = ""
+		handler(context, w, r)
+		return
 	}
 
-	// Mattermost-Plugin-ID can only be set by inter-plugin requests
-	r.Header.Del("Mattermost-Plugin-ID")
+	csrfCheckPassed := false
+	if (session != nil && session.Id != "") && appErr == nil && cookieAuth && r.Method != "GET" {
```

Comment:
At this point, can we confidently know there is a session? (And can we remove these checks?)

Similarly, can we stop relying on the `appErr` being used at L176 and L226 (I think these line numbers are now stale from the comment now on L184 anyway).